src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -725,15 +710,17 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
                 returnType = TypeWithAnnotations.Create(Binder.Compilation.GetSpecialType(SpecialType.System_Void));
             }
 
-            return Binder.GetMethodGroupOrLambdaDelegateType(
-                _unboundLambda.Syntax,
-                returnRefKind,
-                returnType,
+            var finalLambdaSymbol = new LambdaSymbol(
```

Comment:
>new LambdaSymbol [](http://example.com/codeflow?start=36&length=16)

What does this `LambdaSymbol` instance include that is different from the `lambdaSymbol` local? Is it just `returnRefKind` and `returnType`? If so, could we pass those two values directly in addition to the `MethodSymbol`, or simply continue to pass all values separately?

Reply:
> What does this `LambdaSymbol` instance include that is different from the `lambdaSymbol` local? Is it just `returnRefKind` and `returnType`?

Exactly, that's the difference.

> If so, could we pass those two values directly in addition to the `MethodSymbol` rather than creating an additional `LambdaSymbol` and related `ParameterSymbol` instances, etc.?

OK, I can do that: 6b76d8aa8f8db5eaffee6d91f786eb6b18ca81b2. But note that the first `LambdaSymbol` would not always be created, only if there were any parameter attributes or an explicit return type.

> What was incorrect previously?

There was duplication of code with at least one inconsistency of default parameter values handling, resulting in `var lam = ([DefaultParameterValue(3)] int x) => x; lam();` to be working, even though an `[Optional]` attribute was missing (see 5a60ba785c6470a0dc04b899d995c0555e7a8d6f). In another code path (involving method groups instead of lambdas), this bug would not occur. Sharing the code that extracts the information from `MethodSymbol` avoids these inconsistencies.

> could we simply continue to pass all values separately?

Yes, but I wanted to simplify this code as suggested by @AlekseyTs in https://github.com/dotnet/roslyn/pull/64823#discussion_r1002210487.